central/externalbackups/plugins/s3compatible/s3compatible.go

```diff
@@ -0,0 +1,238 @@
+package s3compatible
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
```

Comment:
I think we should leave a few comments here why we are using aws sdk v2 here while we use sdk v1 in `s3.go` (reason being the GCS backwards compatibility). I would also add a comment on why we distinguish between s3 and s3compatible (slight divergence in feature set already and maybe more drift in the future).

Reply:
Makes sense, I'll add a comment 👍

Reply:
I've added a comment. Let me know what you think.